maven-model-builder/src/main/java/org/apache/maven/model/building/DefaultModelBuilder.java

```diff
@@ -852,10 +852,11 @@ private ModelData readParent( Model childModel, ModelSource childSource, ModelBu
                 File pomFile = parentData.getModel().getPomFile();
                 if ( pomFile != null )
                 {
+                    FileModelSource pomFileModel = new FileModelSource( pomFile );
                     ModelSource expectedParentSource = getParentPomFile( childModel, childSource );
 
                     if ( expectedParentSource == null || ( expectedParentSource instanceof ModelSource2
-                        && !pomFile.toURI().equals( ( (ModelSource2) expectedParentSource ).getLocationURI() ) ) )
+                        && !pomFileModel.equals(  expectedParentSource ) ) )
```

Comment:
At first the comparison looked weird, i.e. comparing a Model with a Source. The comparison is good, just need to improve the name of the variables.

Reply:
changed